src/main/scala/fasta.scala

```diff
@@ -145,30 +96,30 @@ case object fasta {
 
     /* Use this method to parse all FASTA values from the `lines` iterator (with possible errors) */
     def parseFasta():
-        Iterator[ Either[ParseDenotationsError, FASTA.Value] ] =
-    new Iterator[ Either[ParseDenotationsError, FASTA.Value] ] {
+        Iterator[ Option[FASTA] ] =
+    new Iterator[ Option[FASTA] ] {
 
       /* If there is one more header, there is one more FASTA value (even if the sequence is empty) */
-      def hasNext: Boolean = lines.hasNext && fasta.header.is(lines.head)
+      def hasNext: Boolean = lines.hasNext && Header.isValid(lines.head)
 
-      def next(): Either[ParseDenotationsError, FASTA.Value] = {
-        val hdr: String = lines.next()
-        val seq: String = cutUntil(fasta.header.is).mkString
+      def next(): Option[FASTA] = {
```

Comment:
this should call hasNext first

Reply:
Fixed in c9ff593. But I have another concern about this iterator:
- it should either return `FASTA` without `Option`, because `hasNext` already checks that the next line is a valid header
- or `hasNext` should do that check

So you either have an Iterator, which reads all lines and returns `None` for chunks which are not a valid FASTA, or you have an iterator which reads only consecutive valid `FASTA` lines (and then it doesn't need to return `Option`).

Reply:
P.S. we have to choose one or another way to modify it, because currently it just won't ever return `None`s.

Reply:
In my opinion `parseFasta: Iterator[Option[FASTA]]` which returns `None` for the chunks that don't contain any valid FASTA header makes more sense. Then we still have 
```scala
parseFastaDropErrors: Iterator[FASTA] = 
  parseFasta.flatten
```

And the other iterator I meantion above, which parses only consecutive valid FASTAs will be defined like this: 
```scala
parseConsecutiveFasta: Iterator[FASTA] = 
  parseFasta.takeWhile(_.isDefined).flatten
```

Reply:
I just realised that currently once you got a line with a valid header, everything after it until the next valid header will be treated as the corresponding sequence: `cutUntil(Header.isValid)`, so most of the above doesn't make much sense. If there is no way to know where a sequence ends and some crap starts, there is no sense to return `Option`. We can only decide to drop or fail on some crap in the beginning (before the first valid header):

```scala
// implemented as it is now, but returns FASTA, not Option
def parseFasta: Iterator[FASTA] = {
  def hasNext: Boolean = lines.hasNext && Header.isValid(lines.head)
  // i.e. is empty if first line is not a valid header
}

// same, but skips crap until the first valid header
def parseFastaSkipCrap: Iterator[FASTA] = {
  cutUntil(Header.isValid)
  parseFasta
}
```

Reply:
OK to https://github.com/ohnosequences/fastarious/pull/25/files#r144604543

Reply:
Done in 7ae79fa